src/Sentry/Internal/Hub.cs

```diff
@@ -342,12 +342,27 @@ public void CaptureTransaction(Transaction transaction)
             {
                 // Apply scope data
                 var currentScope = ScopeManager.GetCurrent();
-                currentScope.Key.Evaluate();
-                currentScope.Key.Apply(transaction);
+                var scope = currentScope.Key;
+                scope.Evaluate();
+                scope.Apply(transaction);
 
                 // Apply enricher
                 _enricher.Apply(transaction);
 
+                if (transaction.IsSampled != false)
+                {
+                    foreach (var processor in scope.GetAllTransactionProcessors())
```

Comment:
We apply event and exception processors in `SentryClient`.  Why did you choose to apply transaction processors in `Hub`?

Reply:
because it needs access to the `scope`. which would be a breaking change in `SentryClient`

Reply:
i would like to keep the approach consistent if u can see a way to do it in a non breaking way?

Reply:
It seems that `CaptureEvent` takes an optional scope parameter, where the default `null` will use the current scope.  Since `CaptureTransaction` doesn't take a `scope` parameter, but `Hub.CaptureTransaction` gets the current scope from the scope manager, that you could also get the current scope in `SentryClient.CaptureTransaction` and evaluate the transaction processors there.

Alternatively, we could just only allow transaction processors to be added on `SentryOptions` globally, and not on a specific scope.  I don't quite understand the use case for adding an event/transaction processor only to a scope anyway.

I must admit I don't quite understand the need for all these layers of indirection we have... Something perhaps to re-evaluate later.

Reply:
@bruno-garcia can u weigh in here? is there any use case for scoped transaction processors? and, historically, what was the use case of scoped EventProcessors and ExceptionProcessors

Reply:
Ping @bruno-garcia